
package NKikimrConfig;
option java_package = "ru.yandex.kikimr.proto";

message TFeatureFlags {
    enum Tribool {
        UNSET = 0;
        VALUE_TRUE = 1;
        VALUE_FALSE = 2;
    }

    optional bool EnableSeparateSolomonShardForPDisk = 1 [default = true];
    optional bool UseForsetiSchedulerByDefaultForPDisk = 2 [default = true];
    optional bool EnableSeparateTrimThreadForPDisk = 3 [default = true];
    optional bool EnableSeparateSubmitThreadForPDisk = 4 [default = true];
    optional bool EnablePerOwnerChunkLimitForPDisk = 5 [default = false];
    optional bool TrimEntireDeviceOnStartup = 6 [default = false];
    optional bool EnableChunkGraceForPDisk = 7 [default = true];
    optional bool AllowConsistentOperationsForSchemeShard = 8 [default = true];
    optional bool EnableSchemeBoard = 9 [default = true]; // deprecated: always true
    optional bool AllowYdbRequestsWithoutDatabase = 10 [default = true];
    optional bool EnableExternalSubdomains = 11 [default = true];
    optional bool AllowRecursiveMkDir = 12 [default = true]; // deprecated: always true
    optional bool AllowHugeKeyValueDeletes = 13 [default = true]; // delete when all clients limit deletes per request
    optional bool SendSchemaVersionToDatashard = 14 [default = true]; // deprecated: always true
    optional bool EnableSchemeBoardCache = 15 [default = true]; // deprecated: always true
    optional bool EnableSystemViews = 16 [default = true];
    optional bool EnableExternalHive = 17 [default = true];
    optional bool UseSchemeBoardCacheForSchemeRequests = 18 [default = true]; // deprecated: always true
    optional bool CompileMinikqlWithVersion = 19 [default = true]; // deprecated: always true
    optional bool ReadTableWithSnapshot = 20 [default = true]; // deprecated: always true
    optional bool ImportantTabletsUseSystemPool = 21 [default = true];
    optional bool EnableOfflineSlaves = 22 [default = true]; // deprecated: always true
    optional bool CheckDatabaseAccessPermission = 23 [default = false];
    optional bool AllowOnlineIndexBuild = 24 [default = true]; // deprecated: always true
    optional bool EnablePersistentQueryStats = 25 [default = true];
    optional bool DisableDataShardBarrier = 26 [default = false];
    optional bool EnablePutBatchingForBlobStorage = 27 [default = true];
    optional bool EnableKqpWideFlow = 28 [default = true]; // deprecated: always true
    optional bool EnableKqpScanQueries = 29 [default = true]; // deprecated: always true
    optional bool EnablePersistentPartitionStats = 30 [default = false];
    optional bool EnableTtlOnIndexedTables = 31 [default = true]; // deprecated: always true
    optional bool AllowUpdateChannelsBindingOfSolomonPartitions = 32 [default = false];
    optional bool DisableLegacyYql = 33 [default = true]; // deprecated: always true
    optional bool EnableDataColumnForIndexTable = 34 [default = true];
    optional bool AllowServerlessStorageBillingForSchemeShard = 35 [default = false];
    optional bool EnableGracefulShutdown = 36 [default = true];
    optional bool EnableDrainOnShutdown = 37 [default = true];
    optional bool EnableConfigurationCache = 38 [default = false];
    optional bool EnableDbCounters = 39 [default = false];
    optional bool EnableClockGettimeForUserCpuAccounting = 40 [default = false];
    optional bool EnableAsyncIndexes = 41 [default = true]; // deprecated: always true
    optional bool AllowStreamExecuteYqlScript = 42 [default = true];
    optional bool EnableKqpScanOverPersistentSnapshot = 43 [default = true]; // deprecated: always true
    optional bool EnableOlapSchemaOperations = 44 [default = true];
    optional bool EnableVPatch = 45 [default = true];
    optional bool EnableMvccSnapshotReads = 46 [default = true]; // deprecated: always true
    optional Tribool EnableMvcc = 47 [default = VALUE_TRUE]; // deprecated: always true
    optional bool EnableSchemeTransactionsAtSchemeShard = 48 [default = true];
    optional bool EnableArrowFormatAtDatashard = 49 [default = false];
    optional bool Enable3x3RequestsForMirror3DCMinLatencyPut = 50 [default = false];
    optional bool EnableBackgroundCompaction = 51 [default = true];
    optional bool EnableArrowFormatInChannels = 52 [default = false];
    optional bool EnableBackgroundCompactionServerless = 53 [default = false];
    optional bool EnableNotNullColumns = 54 [default = true];
    optional bool EnableTtlOnAsyncIndexedTables = 55 [default = true]; // deprecated: always true
    optional bool EnableBulkUpsertToAsyncIndexedTables = 56 [default = true];
    optional bool EnableNodeBrokerSingleDomainMode = 57 [default = true];
    reserved 58; // optional bool EnableKqpSessionActor = 58 [default = true];
    // Dangerous settings we don't want in the public api unless required
    optional bool EnablePublicApiExternalBlobs = 59 [default = false];
    optional bool EnablePublicApiKeepInMemory = 60 [default = false];
    optional bool EnableImplicitScanQueryInScripts = 61 [default = true];
    reserved 62; // EnablePredicateExtractForScanQueries
    optional bool AllowVDiskDefrag = 63 [default = true];
    optional bool EnableAsyncHttpMon = 64 [default = true];
    optional bool EnableChangefeeds = 65 [default = true];
    reserved 66; // EnableKqpScanQueryStreamLookup
    optional bool EnableKqpScanQueryMultipleOlapShardsReads = 67 [default = false];
    reserved 68; // EnablePredicateExtractForDataQueries;
    reserved 69; // optional bool EnableKqpPatternCacheLiteral = 69 [default = false];
    optional bool EnableMoveIndex = 70 [default = true];
    // enable http handle for self termination
    optional bool EnableFailureInjectionTermination = 71 [default = false];
    optional bool EnableChunkLocking = 72 [default = false];
    optional bool EnableNotNullDataColumns = 73 [default = true];
    optional bool EnableGrpcAudit = 74 [default = false];
    reserved 75; // EnableKqpDataQueryStreamLookup
    optional bool EnableBorrowedSplitCompaction = 76 [default = true];
    optional bool EnableChangefeedInitialScan = 77 [default = true];
    reserved 78; // EnableKqpScanQuerySourceRead
    optional bool EnableDynamicNodeAuthorization = 79 [default = false];
    reserved 80; // EnableKqpImmediateEffect
    optional bool EnableDataShardGenericReadSets = 81 [default = false];
    // enable alter database operation to create subdomain's system tablets
    // directly in subdomain's hive
    optional bool EnableAlterDatabaseCreateHiveFirst = 82 [default = false];
    reserved 83; // EnableKqpDataQuerySourceRead
    optional bool EnableSmallDiskOptimization = 84 [default = true];
    optional bool EnableDataShardVolatileTransactions = 85 [default = true];
    optional bool EnableTopicServiceTx = 86 [default = false];
    optional bool EnableLLVMCache = 87 [default = false];
    optional bool EnableExternalDataSources = 88 [default = false];
    optional bool EnableTopicDiskSubDomainQuota = 89 [default = true];
    optional bool EnableSeparationComputeActorsFromRead = 90 [default = false];
    optional bool EnablePQConfigTransactionsAtSchemeShard = 91 [default = false];
    optional bool EnableScriptExecutionOperations = 92 [default = true];
    optional bool EnableImplicitQueryParameterTypes = 93 [default = true];
    optional bool EnableForceImmediateEffectsExecution = 94 [default = false];
    optional bool EnableTopicSplitMerge = 95 [default = false];
    optional bool EnableChangefeedDynamoDBStreamsFormat = 96 [default = true];
    optional bool ForceColumnTablesCompositeMarks = 97 [default = false];
    optional bool EnableSubscriptionsInDiscovery = 98 [default = false];
    optional bool EnableGetNodeLabels = 99 [default = false];
    optional bool EnableTopicMessageMeta = 100 [default = true];
    optional bool EnableIcNodeCache = 101 [default = true];
    optional bool EnableTempTables = 102 [default = false];
    optional bool SuppressCompatibilityCheck = 103 [default = false];
    optional bool EnableUniqConstraint = 104 [default = false];
    optional bool EnableChangefeedDebeziumJsonFormat = 105 [default = false];
    optional bool EnableStatistics = 106 [default = false];
    optional bool EnableUuidAsPrimaryKey = 107 [default = false];
    optional bool EnableTablePgTypes = 108 [default = false];
    optional bool EnableLocalDBBtreeIndex = 109 [default = true];
    optional bool EnablePDiskHighHDDInFlight = 110 [default = false];
    optional bool UseVDisksBalancing = 111 [default = false];
    optional bool EnableViews = 112 [default = false];
    optional bool EnableServerlessExclusiveDynamicNodes = 113 [default = false];
    optional bool EnableAccessServiceBulkAuthorization = 114 [default = false];
    optional bool EnableAddColumsWithDefaults = 115 [ default = false];
    optional bool EnableReplaceIfExistsForExternalEntities = 116 [ default = false];
    optional bool EnableCMSRequestPriorities = 117 [default = false];
    optional bool EnableKeyvalueLogBatching = 118 [default = false];
    optional bool EnableLocalDBFlatIndex = 119 [default = true];
    optional bool ExtendedVDiskCounters = 120 [default = true];
    optional bool ExtendedPDiskSensors = 121 [default = true];
    optional bool EnableDynamicNodeNameGeneration = 122 [default = false];
    optional bool EnableBackupService = 123 [default = false];
    optional bool EnableVolatileTransactionArbiters = 124 [default = false];
    optional bool EnableGraphShard = 125 [default = false];
    optional bool EnableExternalSourceSchemaInference = 126 [default = false];
}
